core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -1424,24 +1425,28 @@ class ReplicaManagerTest {
       metadataCache, mockLogDirFailureChannel, mockProducePurgatory, mockFetchPurgatory,
       mockDeleteRecordsPurgatory, mockElectLeaderPurgatory, Option(this.getClass.getName)) {
 
-      override protected def createAsyncReplicaFetcherManager(metrics: Metrics,
+      override protected def createReplicaFetcherManager(metrics: Metrics,
```

Comment:
should we have unit tests for both async and original fetchermanager?

Reply:
Good call, I've updated the test to test both behaviors.